src/api/data/cache/charthelper/icon_helper.go

```diff
@@ -0,0 +1,142 @@
+package charthelper
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/disintegration/imaging"
+	"github.com/helm/monocular/src/api/swagger/models"
+)
+
+const originalFormat = "original"
+
+type iconFormat struct {
+	name   string
+	width  int
+	height int
+}
+
+var availableFormats = []iconFormat{
```

Comment:
With the current configuration we are generating images that fill the given height and width. For example, for Traefik the output is:

![lTraefik 100 x 100](https://cloud.githubusercontent.com/assets/4056725/22733350/af06f87a-edf1-11e6-83c0-475bf4044376.png)

The problem is the logo is cropped, so some parts of the logo are missing. For image backgrounds is okey because we want to fill it as much as possible. But, for cards we want to show the entire logo. I think we can add another attribute to `iconFormat` called `strategy` to switch between `fill` and `fit` methods:

```go
// Strategy resizing method type
type StrategyType uint8

const(
  FIT StrategyType = iota
  FILL
)

type iconFormat struct {
  name         string
  strategy     StrategyType
  width        int
  height       int
}


var availableFormats = []iconFormat{
  {"100x100", FILL, 100, 100},
  {"300x300", FILL, 300, 300},
  {"100x100fit", FIT, 100, 100},
  {"300x300fit", FIT, 300, 300},
}
```

This parameter allow us to use an `fill` or `fit` strategy to resize and crop the image. The result for the same logo is:

![Traefik 100x100fit](https://cloud.githubusercontent.com/assets/4056725/22733472/3765b13e-edf2-11e6-818e-f3ec9e68f8f7.png)

I know this image doesn't measure 100 x 100px, but `fit` method ensures me that the maximum height or width will be 100px. So, I can center it using `background` properties in the CSS.

We need to update the `processIcon` method to check the strategy:

```go
var processed *image.NRGBA
if format.strategy == FILL {
  processed = imaging.Fill(orig, format.width, format.height, imaging.Center, imaging.Lanczos)
} else {
  processed = imaging.Fit(orig, format.width, format.height, imaging.Lanczos)
}

```